source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,196 @@
+      std::vector<double> local_surface_area_integral;
+      std::vector<double> local_second_invariant_of_strain_rate_and_area;
+      std::vector<double> sorted_local_second_invariant_of_strain_rate;
+
+      const unsigned int n_q_points = quadrature_formula_face.size();
+      const unsigned int n_compositional_fields = this->n_compositional_fields();
+      //std::vector<double> compositional_values(n_q_points);
+
+      typename MaterialModel::Interface<dim>::MaterialModelInputs in(n_q_points,
+                                                                     n_compositional_fields);
+      typename MaterialModel::Interface<dim>::MaterialModelOutputs out(n_q_points,
+                                                                       n_compositional_fields);
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned() && cell->at_boundary())
+          {
+            unsigned int face_idx = numbers::invalid_unsigned_int;
+            bool at_upper_surface = false;
+            {
+              for (unsigned int f=0; f<GeometryInfo<3>::faces_per_cell; ++f)
+                {
+                  if (cell->at_boundary(f) && cell->face(f)->boundary_id() == top_boundary_id)
+                    {
+                      // If the cell is at the top boundary, assign face_idx.
+                      face_idx = f;
+                      at_upper_surface = true;
+                      break;
+                    }
+                }
+
+            }
+            fe_face_values.reinit(cell, face_idx);
+
+            if (at_upper_surface)
+              {
+                in.reinit(fe_face_values,
+                          cell,
+                          this->introspection(),
+                          this->get_solution());
+                //in.requested_properties = MaterialModel::MaterialProperties::viscosity;
+
+                this->get_material_model().fill_additional_material_model_inputs(in,
+                                                                                 this->get_solution(),
+                                                                                 fe_face_values,
+                                                                                 this->introspection());
+
+                this->get_material_model().evaluate(in, out);
+
+                for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    // Calculate the square root of the second moment invariant for the deviatoric strain rate tensor.
+                    //edot_ii = std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[q])))),
+                    //          min_strain_rate);
+
+                    const double local_second_invariant_of_strain_rate = (std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[q])))),min_strain_rate) * fe_face_values.JxW(q));
```

Comment:
Dont multiply with the volume weight of this quadrature point (JxW) here, you are not integrating yet.
```suggestion
                    const double local_second_invariant_of_strain_rate = std::max(std::sqrt(std::fabs(second_invariant(deviator(strain_rate[q])))),min_strain_rate);
```